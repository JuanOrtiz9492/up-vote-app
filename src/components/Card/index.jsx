import React, {Component} from 'react';
import styled from 'styled-components';
import ProgressBar from '../ProgressBar';
import ButtonGroup from '../ButtonGroup';
import Description from './components/Description';
import {ThumbsUpIcon, ThumbsDownIcon} from '../Icon/IconList'
import { cssConstants } from '../../config/cssConstants';

const StyledCardContainer = styled.div`
    display: grid;
    grid-template-rows: 87% 13%;
    width: 533px;
    height: 533px;
    background-image: url(${props => require(`../../assets/img/${props.image}`)});
    @media(max-width: 800px){
        margin: 2em 0;
    }
`

const BottomContainer = styled.div`
    position: relative;
    align-self: flex-end;
    bottom: 3em;
    height: max-content;
    margin: 0 auto;
`
const MidContainer = styled.div`
    width: fit-content;
    position: relative;
    align-self: flex-end;
    display: grid;
    grid-template-columns: 2.6em 1fr;
`

const StyledIcon = styled.div`
    padding: 0.3em;
    margin-top: 0.5em;
    background-color: ${props => props.color};
    height: 2em;
    width: max-content;
`

const {
    fire,
    jungle,
    white
} = cssConstants.colors;

const Card = ({
    id,
    image,
    name,
    lastUpdate,
    aditionalText,
    upVotes,
    downVotes,
    voteHandler,
    }) => {
        const voteTypeHandler = (voteType) => {
            if (voteType === 'upVote') {
                voteHandler(id, ++upVotes, downVotes);
            } else if (voteType === 'downVote') {
                voteHandler(id, upVotes, ++downVotes);
            }
        }
        return (
            <StyledCardContainer image={image}>
                <MidContainer>
                    {upVotes >= downVotes ? 
                        <StyledIcon color={jungle.hex}>
                            <ThumbsUpIcon width={1.8} color={white.hex}/>        
                        </StyledIcon> : 
                        <StyledIcon color={fire.hex}>
                            <ThumbsDownIcon width={1.8} color={white.hex}/>        
                        </StyledIcon>
                    }
                    <Description name={name} lastUpdate={lastUpdate} aditionalText={aditionalText}/>
                    <div/>
                    <ButtonGroup voteHandler={voteTypeHandler}/>
                </MidContainer>
                <BottomContainer>
                    <ProgressBar value={upVotes} maxValue={upVotes+downVotes}/>
                </BottomContainer>
            </StyledCardContainer>
        )}

export default Card;
