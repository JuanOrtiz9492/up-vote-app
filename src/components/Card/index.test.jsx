import React from 'react';
import { render } from '@testing-library/react';
import Card from '.';

describe('Card Component', () => {

    it('should Match the Snapshot', () => {
        const { container } = render(<Card image={'kanye-west.jpg'} />);
        expect(container).toMatchSnapshot();
      })
})

