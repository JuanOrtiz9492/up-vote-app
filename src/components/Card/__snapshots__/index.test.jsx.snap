// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should Match the Snapshot 1`] = `
<div>
  <div
    class="sc-fzXfLV iGSUfY"
  >
    <div
      class="sc-fzXfLX iPgGDN"
    >
      <div
        class="sc-fzXfMv enHMqV"
        color="#ffad1d"
      >
        <svg
          class="sc-AykKD bCBIIE"
          color="#ffffff"
          width="1.8"
        >
          thumbs-down.svg
        </svg>
      </div>
      <section
        class="sc-fzXfLR hupksK"
      >
        <h2
          class="sc-fzXfLS jcoINV"
        />
        <p
          class="sc-fzXfLT iBbqek"
        />
        <p
          class="sc-fzXfLU ctXhhA"
        />
      </section>
      <div />
      <div
        class="sc-fzXfLQ bexvcS"
      >
        <button
          class="sc-fzXfLP kOVuKz jungle"
          color="#ffffff"
        >
          <svg
            class="sc-AykKC cVkhbB"
            color="#ffffff"
            width="1.2"
          >
            thumbs-up.svg
          </svg>
        </button>
        <button
          class="sc-fzXfLP kOVuKz fire"
          color="#ffffff"
        >
          <svg
            class="sc-AykKD SFEtK"
            color="#ffffff"
            width="1.2"
          >
            thumbs-down.svg
          </svg>
        </button>
        <button
          class="sc-fzXfLP kOVuKz"
          color="#ffffff"
        >
          vote now
        </button>
      </div>
    </div>
    <div
      class="sc-fzXfLW biiXZr"
    >
      <div
        class="sc-AykKJ fNCYSY progress-bar"
      >
        <div
          class="sc-AykKK gPvMTr left"
        >
          <svg
            class="sc-AykKC iUTaVw"
            color="#ffffff"
            width="1.5"
          >
            thumbs-up.svg
          </svg>
          <span
            class="sc-fzXfLO iMHASE"
          >
            NaN%
          </span>
        </div>
        <div
          class="sc-AykKK gPvMTr right"
        >
          <span
            class="sc-fzXfLO iMHASE"
          >
            NaN%
          </span>
          <svg
            class="sc-AykKD gSoynF"
            color="#ffffff"
            width="1.5"
          >
            thumbs-down.svg
          </svg>
        </div>
        <progress
          class="sc-AykKI cRdaSl"
          max="NaN"
          value="50"
        />
      </div>
    </div>
  </div>
</div>
`;
